### PR TITLE
1463 note schema update

### LINKED ISSUE #1463: Update Note data structure and controls
While importing data it became apparent that some of the required note fields are not available. Thus only make the Note text required.

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #1463

Only makes the `text` Note property required.

For testing: check that this doesn't break the UI.

### CODE REVIEW

File: client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx [hunk 1]
```diff
@@ -63,7 +64,7 @@ const NotesComponent = (props: ArrayControlCustomProps) => {
 
   const options = props.uischema.options;
 
-  const inputData = data ?? [];
+  const inputData = (data as unknown as NoteSchema[]) ?? [];
```
Comment (CarlosNZ) [inline client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx:67]: Why the double `as`? It compiles fine with just `as NoteSchema[]`.

File: client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx [hunk 2]
```diff
@@ -201,12 +196,12 @@ const NoteComponent = (props: ControlProps) => {
...
         error
           ? undefined
-          : {
+          : ({
               text,
               authorId,
               authorName,
               created,
-            }
+            } as NoteSchema)
```
Comment (CarlosNZ) [inline client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx:204]: `as NoteSchema` isn't necessary here.

**Review by CarlosNZ — APPROVED** [review]
LGTM! Just a couple of minor Typescript nits.

File: client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx (hunk 2 above)
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx:204]: handleChange only takes an any, that change was just to keep sure this has the shape of a NoteSchema

File: client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx]: thanks! I removed it, I think I made a different change to make this unnecessary...

**Pull request merged**